### PR TITLE
Refresh the Spotify token when needed

### LINKED ISSUE #21: The overlay stops working overtime.
The overlay seems to stop working over time. The way I fix this is by going back onto your website and re-adding the link into my browser source on obs.

### LINKED ISSUE #28: How To Self-Host?
Hey, first of all, your overlay is very nifty and makes the stream cool

I noticed you have a 60-minute expiration time for the tokens, and I have to regenerate the link 2 or 3 times during the stream and I don't even notice it sometimes, I know removing the time limit would be catastrophic for the API application, so could you please help me set this up with my own Spotify dev app and deploy it from my machine and completely remove the token thing.

### LINKED ISSUE #39: Overlay Stops working after 60mins
Ive already seen a post below, but i thought i repost the issue,... every 60 minutes it stops wokring and i have to rel-authorize thru you website. Can this be fixed please?

Kind Regards 
Philipp

### PR DESCRIPTION
**Pull request by joaopms**
**TL;DR: The overlay continues working after 60 minutes.**

This project was with using [Spotify's Implicit Grant flow](https://developer.spotify.com/documentation/web-api/tutorials/implicit-flow), which only worked for 60 minutes. After that, the access token used to fetch the currently playing song expires and the overlay stops updating.

By migrating to the [Authorization Code with PKCE flow](https://developer.spotify.com/documentation/web-api/tutorials/code-pkce-flow), we're now able to refresh the access token for as long as we need, meaning the overlay continues updating even after 60 minutes.

Fixes #21 and fixes #28 and fixes #39 

---

Dependencies had to be updated to run on the latest Node.js LTS (version 20), which _should_ close all of @app/dependabot PRs

### CODE REVIEW

**Comment by joaopms** [comment]
Even though this PR hasn't been merged, the [repo with all the fixes is public](https://github.com/joaopms/spotify-now-playing-overlay) (including #38) and deployed at https://joaopms.github.io/spotify-now-playing-overlay.
